packages/design-system/src/components/Button/Button.js

```diff
@@ -26,19 +44,21 @@ const Button = React.forwardRef((props, ref) => {
     ariaExpanded,
     className,
     type: htmlType,
+    kind,
+    iconPosition,
     ...buttonProps
   } = props;
 
   const isDisabled = disabled || loading;
   const type = submit ? 'submit' : htmlType || 'button';
-  let buttonType = null;
+  const isValidKind = kind && acceptedKinds.includes(kind);
+  const deprecatedKind = getDeprecatedKind(primary, destructive, secondary);
+  const buttonKind = (isValidKind && kind) || deprecatedKind || null;
 
-  if (primary) {
-    buttonType = 'primary';
-  } else if (destructive) {
-    buttonType = 'destructive';
-  } else if (secondary) {
-    buttonType = 'secondary';
+  if (deprecatedKind) {
+    noticeAboutDeprecation(
+      `deprecated prop '${deprecatedKind}' - please use 'kind' prop instead`
```

Comment:
shouldn't you add a component name to the message to make it easier for devek to find it and fix?

Reply:
6de75d239c41beca9ffd5f3ee6d5b65204383df7